Johanna Eriksson Portfolio

joheri1 opened the pull request:
## Johannas Portfolio [link](https://johanna-eriksson-portfolio.netlify.app/)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
  - Introduced a comprehensive README file with project setup instructions for the React portfolio.
  - Added a new `index.html` file as the main entry point for the application.
  - Implemented several new components, including `Hero`, `Footer`, `CardSection`, and `TextSection`, enhancing the portfolio's layout.
  - Created a `Button` component for consistent button styling and functionality.
  - Added social media integration with a `SocialIcons` component.
  - Introduced a structured `projects.json` file for project data management.

- **Improvements**
  - Established responsive design through CSS for various components and sections.

- **Chores**
  - Added configuration files for ESLint and Vite to improve development workflow.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
Add new folder for components

Commit:
Initialize using Vire and install my-portfolio template. Add instructions on how to do this in the README file. Add a folder for Components.

Commit:
Add global styles in index.css

Commit:
Start adding the structure. Create folders for components, create components for the buttons, add App.css, update App.jsx to contain relevant code. Add Hero.css and Hero.jsx.

Commit:
Add components for section, image, card, button, typography, tag, and grid. Create css files for each component. Set the structure by adding the components to different folders.

Commit:
Start writing the README file

Commit:
Add icons to the icons folder

Commit:
”To the Section folder, add TextSection.jsx, CardSection.jsx, and Footer.jsx. Create CSS files for each component, and add the components in own folders in the section folder. Add basic code to the section components.

Commit:
Add unsaved changes, if any

Commit:
Mount components to App.jsx. Add some first lines of code to the components Hero, TextSection, and CardSection.

Commit:
Rename component TextSection CSS, that mistakingly was given the wrong name

Commit:
Update paths, etc to make the code work. Import the Footer component to App.jsx, not FooterSection.

Commit:
Fix code for the section since it is using a mix of children and explicit props for passing content. Add information to the README file

Commit:
Change px to rem on the font sizes

Commit:
Changed px to rem on each components css file

Commit:
Correct variable names for the h1 and h2 fonts in the sections.

Commit:
Add image to Assets.

Commit:
Add styling to Hero. Add social icons to hero incl css

Commit:
Add more styling to Hero

Commit:
Add styling to Tech text section. Update section comments in the finished components. Add image for footer.

Commit:
Complete the Footer component and add styling to it. Create a new component for Icons. Update Footer and Hero components to import the Icon.jsx. Update README.

Commit:
Save any unsaved changes

Commit:
Add more default data in the project.json and move the json file to within the src folder. Import json data to the ProjectCard. Add hard coded data for the ArticleCard.

Commit:
Update Button components. Import the Button components to the CardSection. Add Content to the project.json. Add more styling to CardSection.

Commit:
Add descriptions to the components

Commit:
Delete second, unused README file. Update the correct README

Commit:
Fix file path to Typography.css

Commit:
Correct paths to Icon components

Commit:
Correct the paths to Icons in the Hero and Footer components

Commit:
Add more styling to CardSection

Commit:
Force Netlify redeploy

Commit:
Rename Icons to SocialIcons to trigger a change

Commit:
Correct path to Button.css

Commit:
Change how props are being passed in the ProjectCard to a heading prop to display the h3 heading

Commit:
Added more info to the readme file.

Commit:
Save any unsaved changes

Commit:
Add global responsive styling for fonts and font-sizes for Tablet, Desktop and Mobile. Start adding responsive styling to the components

Commit:
Change margin for body text in skills columns

Commit:
Add styling to media queries for Mobile. Fix syntax error in Global styling for font h2 for mobile

Commit:
Add styling to Hero

Commit:
Add styling for media query Tablet

Commit:
Add more styling to Media queries css

Commit:
Update README file

Commit:
Delete CSS file for responsiveness since it is not used. Update comments in the components

Commit:
Add gap to icons in Hero for mobile

Commit:
Add font weight to the fonts in the global styling

joheri1 committed:
Update README.md

Update README.

JennieDalgren reviewed (changes requested):
Great job with this project!
You have reach almost all requirements but there are some small things that differs in your version from the design file.
Take a look at how you handle the spacing of your projects, because they stretch out too much. And on a larger screen it would look like this:
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/9dc5420d-2986-46b4-9112-42055ff4cd51">

Commit:
Update CardSection.css by giving the cards-container a max-width of 1200px to prevent project and article cards som stretching too much.

Commit:
Updated local README to match GitHub version

Commit:
Update Footer.css by giving the footer-container a max-width of 1200px to prevent image and contact info from stretching too far apart. Also add margin 0 auto to center the section, change justify-content from space-evenly to center, and add align-self: stretch.

joheri1 commented:
Thanks Jennie for notifying me!

I've updated the CardSection.css by adding a max-width of 1200px, and I
also updated the Footer.css to fix the large gap issue there as well.

Kindest regards / Med vänlig hälsning,
*====================*
*Johanna Eriksson*
Schibsted Enterprise Tech

Västra Järnvägsgatan 21
105 17 Stockholm

***@***.*** ***@***.***>



On Thu, 7 Nov 2024 at 20:05, Jennie Dalgren ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Great job with this project!
> You have reach almost all requirements but there are some small things
> that differs in your version from the design file.
> Take a look at how you handle the spacing of your projects, because they
> stretch out too much. And on a larger screen it would look like this:
> image.png (view on web)
> <https://github.com/user-attachments/assets/9dc5420d-2986-46b4-9112-42055ff4cd51>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Technigo/project-portfolio/pull/393#pullrequestreview-2421894616>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKKOL7L3H3XNPJ6BMQZRJYLZ7O2YBAVCNFSM6AAAAABRDE7UQSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMRRHA4TINRRGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Anna2024WebDev commented on my-portfolio/src/components/sections/card-section/CardSection.css:
The card section with projects and articles (my journey) is cut off on  does not seem to work on tablet and mobile. It is not scroll able. Suggestion is to add column and overflow-x:auto to the grid so it is scrollable. We did the same in our design project last week so I know it works :-) 

.cards-container {
    grid-auto-flow: column;     /* Cards flow horizontally */
    grid-template-rows: 1fr;     /* To ensure a single row layout */
    overflow-x: auto;                /* Enables horizontal scrolling */**
}

.project-card, .article-card {
    min-width: 250px;              /* Sets minimum width for each card for scrollable layout */**
}

Anna2024WebDev commented on my-portfolio/README.md:
Very good comments in the readme file! I really feel the struggle and I can relate to most of the problems regarding this project week. It is also good to have when looking back on what you have created and learned. 🌟

Anna2024WebDev commented on my-portfolio/src/components/sections/card-section/CardSection.jsx:
Suggestion for improvement;

Break out the article content to a separate json file (as you have did with the projects) and import it to the card section component.

Anna2024WebDev commented on my-portfolio/src/components/sections/text-section/TextSection.jsx:
Suggestion of improvement;

Add a skills.json file instead of hardcoding it here. Will make the code cleaner and easier to maintain.

Anna2024WebDev reviewed (commented):
Great job on the portfolio! You have followed the design to the letter! 🌟👏

Some suggestions of improvements (also see detailed description in the separate comments)

- Break out the article/journey and skills data into separate json files for cleaner code and maintainability.
- The card section with projects and articles is cut off and not scrollable on mobile and tablet. Suggestions to update the css for cards-container so it allow horizontal scrolling.

HIPPIEKICK reviewed (changes requested):
Please make all your scrollable sections actually scrollable as it looks broken now. Also, check how your page looks on all sizes (e.g. the footer doesn't fit in tablet width).

<img width="800" alt="Skärmavbild 2024-12-09 kl  11 39 48" src="https://github.com/user-attachments/assets/d4715f93-232c-41ab-ac2b-71978a8df74b">

<img width="542" alt="Skärmavbild 2024-12-09 kl  11 39 23" src="https://github.com/user-attachments/assets/c8dfa2f5-3015-4474-a923-147441dde239">

Commit:
Add scrolling to card-container in the CardSection.css

Commit:
Add more styling to CardSection.css to make the scrolling look ok. Add webkit-box to article-card body-text to limit the number of lines to 3, and add ellisis (three dots) as text-overflow

Commit:
Make the article-card h3 to have a height of two rows. Add ellipsis if text is longer.

Commit:
Restore tablet responsiveness for Footer component

joheri1 commented:
Thank Matilda for your feedback!

I apologize for the misunderstanding regarding the cards. I believe I took
the instruction a bit too literally when you said we should try to copy the
design as much as possible. I somehow assumed that we would fix the cards
later when we had the correct data. I’ve now addressed this, and the cards
are scrollable :)

I’ve also restored the responsiveness for the tablet layout in the footer,
somehow, it got removed. I also corrected the screen size breakpoint, which
was set incorrectly for some reason.

Thanks again!

Kindest regards / Med vänlig hälsning,
*====================*
*Johanna Eriksson*
Schibsted Enterprise Tech

Västra Järnvägsgatan 21
105 17 Stockholm

***@***.*** ***@***.***>



On Mon, 9 Dec 2024 at 11:41, Matilda Brunemalm ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Please make all your scrollable sections actually scrollable as it looks
> broken now. Also, check how your page looks on all sizes (e.g. the footer
> doesn't fit in tablet width).
> Skarmavbild.2024-12-09.kl.11.39.48.png (view on web)
> <https://github.com/user-attachments/assets/d4715f93-232c-41ab-ac2b-71978a8df74b> Skarmavbild.2024-12-09.kl.11.39.23.png
> (view on web)
> <https://github.com/user-attachments/assets/c8dfa2f5-3015-4474-a923-147441dde239>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Technigo/project-portfolio/pull/393#pullrequestreview-2488410405>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKKOL7LGS7G5XIKFYHWK2JL2EVXT7AVCNFSM6AAAAABRDE7UQSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIOBYGQYTANBQGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

HIPPIEKICK reviewed (approved):
Great job fixing this up Johanna! Sorry for any misunderstanding ✨

joheri1 committed:
Update README.md

Update link to project

joheri1 committed:
Update Footer.jsx
